Draft: Torizon 7 Stable release

andreriesco opened the pull request.

Commit:
Dockerfiles: Change base containers tag to 4 on Toradex supported

Change base containers tag to 4, which are the base containers for
Torizon OS 7, on Toradex supported templates.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Dockerfiles: Change base containers tag to 4 on Community supported

Change base containers tag to 4, which are the base containers for
Torizon OS 7, on Community supported templates.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Dockerfiles: Change base containers tag to 4 on Slint templates

Change base containers tag to 4, which are the base containers for
Torizon OS 7, on Slint templates.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cppdbg: Add engineLogging true on cppdbg of launch.json

Sometimes an error on the execution of the application may not be
shown on the Debug Console, so it may be interesting to get the full
log by setting this "engineLogging" variable to true. However, it prints
a lot of stuff on Debug Console, so if the user wants a smaller log he
can set it to false.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
checkSingleProjectsConflicts: Fix

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco commented on rustSlint/Dockerfile:
```diff
@@ -1,10 +1,10 @@
 ##
-ARG CROSS_SDK_BASE_TAG=3.3.1-bookworm-1.7.2
-ARG BASE_VERSION=3.3.1
+ARG CROSS_SDK_BASE_TAG=4.0.0-1.8.0
```
Hello @tronical,
I am updating the Slint templates to the Torizon OS 7 (which are tag 4), but I am having the following problem on them:
```
[+] Running 1/1
 ✔ Container torizon-rustslinttorizon7-1  Created                                                                  0.1s 
Attaching to rustslinttorizon7-1
rustslinttorizon7-1  | Hello Torizon!
rustslinttorizon7-1  | slint linuxkms backend: unrecognized renderer skia, falling back default
rustslinttorizon7-1  | Using Skia OpenGL renderer
rustslinttorizon7-1  | Rendering at 1920x1080
```
Do you know where may be the issue? Should I update the line `ENV SLINT_BACKEND=linuxkms-skia`?

Obs. It shows the application on the screen, however the mouse doesn't work (is it supposed to?)

tronical replied:
> Hello @tronical,
> 
> I am updating the Slint templates to the Torizon OS 7 (which are tag 4), 

Awesome, thanks :)

> but I am having the following problem on them:
> 
> ```
> 
> [+] Running 1/1
> 
>  ✔ Container torizon-rustslinttorizon7-1  Created                                                                  0.1s 
> 
> Attaching to rustslinttorizon7-1
> 
> rustslinttorizon7-1  | Hello Torizon!
> 
> rustslinttorizon7-1  | slint linuxkms backend: unrecognized renderer skia, falling back default
> 
> rustslinttorizon7-1  | Using Skia OpenGL renderer
> 
> rustslinttorizon7-1  | Rendering at 1920x1080
> 
> ```
> 
> Do you know where may be the issue? Should I update the line `ENV SLINT_BACKEND=linuxkms-skia`?

Yes, you can either do that or (simpler) remove the environment variable from the docker file altogether. It's not necessary to set it anymore as enabling the skia renderer feature at compile time will make it a preferred choice at runtime.

> 
> Obs. It shows the application on the screen, however the mouse doesn't work (is it supposed to?)

If you plug in a mouse via USB-A it should show once you start moving it.

andreriesco replied:
Hello, I tested here and it shows on the screen on arm64 (without the mouse working, not sure why), but on arm32 device (more specifically a colibri-imx6), the application is failing, showing the following message:
```
[+] Running 1/1
 ✔ Container torizon-rustslinttorizon7-1  Created                                                           0.6s 
Attaching to rustslinttorizon7-1
rustslinttorizon7-1  | slint linuxkms backend: unrecognized renderer skia, falling back default
rustslinttorizon7-1  | Hello Torizon!
rustslinttorizon7-1  | Using Skia OpenGL renderer
rustslinttorizon7-1  | Rendering at 800x480
rustslinttorizon7-1  | thread 'main' panicked at /cargo/registry/src/index.crates.io-6f17d22bba15001f/drm-0.12.0/src/control/mod.rs:360:9:
rustslinttorizon7-1  | assertion failed: (has_modifier && modifier.is_some()) || (!has_modifier && modifier.is_none())
rustslinttorizon7-1  | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
rustslinttorizon7-1 exited with code 101
```

andreriesco replied:
Hello, I removed the line `ENV SLINT_BACKEND=linuxkms-skia` and tested here and it shows on the screen on arm64 (without the mouse working, not sure why), but on arm32 device (more specifically a colibri-imx6), the application is failing, showing the following message:
```
[+] Running 1/1
 ✔ Container torizon-rustslinttorizon7-1  Created                                                           0.5s 
Attaching to rustslinttorizon7-1
rustslinttorizon7-1  | Hello Torizon!
rustslinttorizon7-1  | Using Skia OpenGL renderer
rustslinttorizon7-1  | Rendering at 800x480
rustslinttorizon7-1  | thread 'main' panicked at /cargo/registry/src/index.crates.io-6f17d22bba15001f/drm-0.12.0/src/control/mod.rs:360:9:
rustslinttorizon7-1  | assertion failed: (has_modifier && modifier.is_some()) || (!has_modifier && modifier.is_none())
rustslinttorizon7-1  | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
rustslinttorizon7-1 exited with code 101
```
As it was using `linuxkms`, it the `weston` container was not been initiated. Should we use the `weston` container now?

tronical replied:
This particular issue happens only
With some GPUs (it's not arm64 vs 43 related). For example I recently saw it on the Mali of the toradex imx95 Evk.

I've got a fix for that and it will be in Slint 1.9.0 in December.

Not sure how to best proceed.

andreriesco replied:
Ok, since December is close to now, I think it makes sense to release the Slint templates using Slint 1.9.0, with the fix. Also, this will give us time to do [the change regarding the base Slint containers](https://github.com/torizon/vscode-torizon-templates/issues/271), but moving the `slint-sdk` to the `Dockerfile.sdk` of the container or by moving it to the [torizon containers](https://github.com/torizon/torizon-containers) (we talked with the people responsible for the Torizon containers in the team and they told us that the repo is open for contributions and adding the `slint-sdk` container there is also a possibility).

tronical replied:
Sounds good. If you want to release an update for Torizon OS 7 before that, I think it would be okay with this being a known issue, as it only affects certain GPUs.

andreriesco replied:
On Torizon OS 7 a new flag signalizing that a given template is incomplete/has known issues will be added. So since it only affects certain GPUs we can add it to the release, with this flag set as true. I will open an issue to keep track of the problem

andreriesco replied:
Hello @tronical,
Is Slint 1.9 available?

tronical replied:
We're a few days away from the release :( next week.

Regarding the containers, how can I test torizon 7? (Context: https://github.com/torizon/vscode-torizon-templates/issues/271#issuecomment-2492095517 )

andreriesco replied:
Torizon 7 is available both on easy installer and on Torizon Cloud, so just updating the board with the quarterly release should be fine. About the containers, just changing the tags of the base containers to 4, like [here](https://github.com/torizon/vscode-torizon-templates/pull/268/commits/3b9943f6cea0fd2226b85d0b53215068a48112dc), should be fine with it.

Commit:
slint: Remove SLINT_BACKEND flag

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Drop all community contributions from Matheus Castello

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby commented:
/me as community contributor are dropping my contributions, I do not have plans to maintain this, if you @andreriesco would like to maintain or the Toradex leverage this as Toradex supported is Toradex call.

kassane commented:
@andreriesco based on `torizon-7-stable` branch - zig updated (patch)

<details>
<summary>zig-patch</summary>

```diff
diff --git a/zigConsole/.gitignore b/zigConsole/.gitignore
index a1e689d..22be779 100644
--- a/zigConsole/.gitignore
+++ b/zigConsole/.gitignore
@@ -1,3 +1,4 @@
-zig-*/
+.zig-cache/
+zig-out/
 
-executable-*
\ No newline at end of file
+executable-*
diff --git a/zigConsole/Dockerfile b/zigConsole/Dockerfile
index b258da1..7c14126 100644
--- a/zigConsole/Dockerfile
+++ b/zigConsole/Dockerfile
@@ -22,7 +22,7 @@ FROM torizon/cross-toolchain-${IMAGE_ARCH}:${SDK_BASE_VERSION} AS build
 ARG IMAGE_ARCH
 ARG COMPILER_ARCH
 ARG APP_ROOT
-ENV ZIG_PATH /zig/0.11.0/files
+ENV ZIG_PATH /zig/0.13.0/files
 
 # __deps__
 RUN apt-get -q -y update && \
@@ -37,22 +37,22 @@ RUN apt-get -q -y update && \
     rm -rf /var/lib/apt/lists/*
 # __deps__
 
-# Install Zig 0.11.0 (uses LLVM-16.0.6)
+# Install Zig 0.13.0 (uses LLVM-18.1.6)
 RUN if [ "$IMAGE_ARCH" = "arm64" ] ; then \
         wget -q https://github.com/kassane/zigup/releases/download/2023_09_29/zigup.ubuntu-latest-aarch64.zip && \
         unzip zigup.ubuntu-latest-aarch64.zip -d /usr/bin && \
         chmod +x /usr/bin/zigup && \
-        zigup --install-dir /zig 0.11.0 ; \
+        zigup --install-dir /zig 0.13.0 ; \
     elif [ "$IMAGE_ARCH" = "arm" ] ; then \
         wget -q https://github.com/kassane/zigup/releases/download/2023_09_29/zigup.ubuntu-latest-armv7a.zip && \
         unzip zigup.ubuntu-latest-armv7a.zip -d /usr/bin && \
         chmod +x /usr/bin/zigup && \
-        zigup --install-dir /zig 0.11.0 ; \
+        zigup --install-dir /zig 0.13.0 ; \
     elif [ "$IMAGE_ARCH" = "amd64" ] ; then \
         wget -q https://github.com/kassane/zigup/releases/download/2023_09_29/zigup.ubuntu-latest-x86_64.zip && \
         unzip zigup.ubuntu-latest-x86_64.zip -d /usr/bin && \
         chmod +x /usr/bin/zigup && \
-        zigup --install-dir /zig 0.11.0; \
+        zigup --install-dir /zig 0.13.0; \
     fi
 
 RUN chmod +x ${ZIG_PATH}/zig
diff --git a/zigConsole/Dockerfile.sdk b/zigConsole/Dockerfile.sdk
index 6e9fce7..d92ddcc 100644
--- a/zigConsole/Dockerfile.sdk
+++ b/zigConsole/Dockerfile.sdk
@@ -7,7 +7,7 @@ ARG IMAGE_ARCH=
 ##
 # Base container version
 ##
-ARG SDK_BASE_VERSION=0.11.0
+ARG SDK_BASE_VERSION=0.13.0
 
 ##
 # Directory of the application inside container
diff --git a/zigConsole/build.zig b/zigConsole/build.zig
index 3077636..d62018f 100644
--- a/zigConsole/build.zig
+++ b/zigConsole/build.zig
@@ -1,5 +1,4 @@
 const std = @import("std");
-const Path = std.Build.LazyPath;
 
 pub fn build(b: *std.Build) void {
     const target = b.standardTargetOptions(.{});
@@ -7,7 +6,7 @@ pub fn build(b: *std.Build) void {
 
     const binary = b.addExecutable(.{
         .name = "__change__",
-        .root_source_file = Path.relative("src/main.zig"),
+        .root_source_file = b.path("src/main.zig"),
         // or (implicit LazyPath struct, w/ path field)
         // .root_source_file = .{ .path = "src/main.zig" },
         .target = target,
@@ -34,7 +33,10 @@ pub fn build(b: *std.Build) void {
     b.installArtifact(binary);
 
     // overwrite default output
-    b.resolveInstallPrefix(b.fmt("zig-out/{s}/{s}", .{ @tagName(target.getCpuArch()), @tagName(optimize) }), .{});
+    b.resolveInstallPrefix(b.fmt("zig-out/{s}/{s}", .{
+        @tagName(binary.rootModuleTarget().cpu.arch),
+        @tagName(optimize),
+    }), .{});
 
     // This *creates* a Run step in the build graph, to be executed when another
     // step is evaluated that depends on it. The next line below will establish
diff --git a/zigConsole/src/main.zig b/zigConsole/src/main.zig
index 86bb580..7afb02e 100644
--- a/zigConsole/src/main.zig
+++ b/zigConsole/src/main.zig
@@ -2,8 +2,8 @@ const std = @import("std");
 const log = std.log.scoped(.toradex);
 
 // override the std implementation
-pub const std_options = struct {
-    pub const log_level = .info;
+pub const std_options = .{
+    .log_level = .info,
 };
 pub fn main() void {
     log.info("Hello {s}!", .{"Torizon"});
```
</details>

Commit:
deprecated templates: Update info about it

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
nodeJSTypeScript: Changing the maintainer (still a Community template)

This was dropped by the maintainer. Torizon has interest on assuming the
maintainership of this template. However, it will still be maintained as
a Community template (meaning it is not guaranteed to be working
always).

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
rustConsole: Deprecating it for Torizon OS 7

Deprecating it for Torizon OS 7, since the maintainer will no longer be
able to maintain it. #272

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
python3Pyside2QML: Deprecating it for Torizon OS 7

Deprecating this template, as it will be replaced by a PySide6 (Qt6)
template in the future.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cConsole:Deprecating it for Torizon OS 7(move to C/C++ Console template)

Deprecating it for Torizon OS 7, since the maintainer will no longer be
able to maintain it. #273
Also, C Console template is not needed, as the C/C++ Console template
can also be used with C (information explaining how to use GCC instead
of G++ was added to to the `.doc/README.md` of the C/C++ Console
template).

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
zigConsole: Apply patch to bump it to zig v.13.0

Related-to: #274

Commit:
templates.json: Update status and custom_message for Torizon 7

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cpp/cmakeConsole:Remove Dockerfile.sdk on update.json (not needed there)

No need for Dockerfile.sdk on update.json, it does this check in the
projectUpdate.ps1 script. Also, update cpp/cmakeConsole info on
templates.json.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
dotnetAvalonia: Make it Toradex supported

As it is the prefered .NET GUI of the people who use the Torizon IDE
extension, now it will become Toradex supported.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
dotnetAvaloniaFrameBuffer: Make it Toradex supported

As Avalonia is the prefered .NET GUI of the people who use the Torizon
IDE extension, now it will become Toradex supported. Therefore, it's
framebuffer version is now alos Toradex supported.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
genericTemplate: Add debug task, for multi-container project

Add the debug task, on launch.json, that call the
run-container-torizon-release-<arch>. They are needed to allow this
to be used with other projects that have debug, on a multi-container
project.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
projectUpdater: Migration to Python shell scripts

projectUpdater.ps1: Add the mechanism to remove the powershell scripts
and execute the new projectUpdater.xsh, on updgrade to Torizon OS 7.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented on aspnetBlazor/Dockerfile:
```diff
@@ -7,7 +7,7 @@ ARG IMAGE_ARCH=
 # Base container version
 ##
-ARG BASE_VERSION=4.0.0
+ARG BASE_VERSION=4.2.0
```
This conflicts with 3b9943f6cea0fd2226b85d0b53215068a48112dc
Please drop 3b9943f6cea0fd2226b85d0b53215068a48112dc

andreriesco replied:
ok, will fixup the commit

microhobby commented on cConsole/Dockerfile:
```diff
@@ -1,5 +1,5 @@
-ARG CROSS_SDK_BASE_TAG=3.3.1
-ARG BASE_VERSION=3.3.1-bookworm
+ARG CROSS_SDK_BASE_TAG=4.0.0
+ARG BASE_VERSION=4.0.0
```
There is a lot of changes on community templates that will be dropped, this is right?

andreriesco replied:
Actually they are there because the commit with the update is before the one with the drop of the template, so in the end it's ok. I wil double-check it to see if everything has been properly dropped

microhobby commented on cConsole/.vscode/launch.json:
```diff
@@ -38,6 +38,9 @@
             "externalConsole": false,
             "internalConsoleOptions": "openOnSessionStart",
+            "logging": {
+                "engineLogging": true,  // set this to false for a smaller log on the debug console
```
Although this may seem informative, having comments in JSON files is not recommended as other tools may have difficulty parsing them. VS Code that has a special parser for JSON with comments.

andreriesco replied:
Sounds good, I will remove it

andreriesco replied:
Actually I am looking at the JSON files which have comments, they are pretty much all related to VSCode (almost all are at `.vscode/launch.json`), and some of the comments are even added by VSCode, like this task on `aspnetBlazor` for example:
```
        {
            // Use IntelliSense to find out which attributes exist for C# debugging
            // Use hover for the description of the existing attributes
            // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
            "name": "Local Debug",
            "type": "coreclr",
            "request": "launch",
            "preLaunchTask": "build-debug-local",
            // If you have changed target frameworks, make sure to update the program path.
            "program": "${workspaceFolder}/bin/Debug/net8.0/__change__.dll",
            "args": [],
            "cwd": "${workspaceFolder}",
            "stopAtEntry": false,
            // Enable launching a web browser when ASP.NET Core starts. For more information: https://aka.ms/VSCode-CS-LaunchJson-WebBrowser
            "serverReadyAction": {
                "action": "openExternally",
                "pattern": "\\bNow listening on:\\s+(https?://\\S+)"
            },
            "env": {
                "ASPNETCORE_ENVIRONMENT": "Development"
            },
            "sourceFileMap": {
                "/Views": "${workspaceFolder}/Views"
            }
        },
```
So in this files that are just related to VSCode I think we can keep the comments, as they can be helpful for the user.

microhobby replied:
you are right, as I said this sounds informative. But we have also to think when these are used on scripts and outside VS Code. So, because of this, I'm more oriented to follow what all the parsers out there follow.

andreriesco replied:
But is `launch.json` used anywhere outside VSCode? I tend to like the comments on this one if it's used just within VSCode

andreriesco replied:
Removed it in https://github.com/torizon/vscode-torizon-templates/pull/268/commits/1d287e9df285cbba9ca8248fa307a8549f54a0a8

microhobby commented on cppConsole/.vscode/launch.json:
```diff
@@ -26,6 +26,53 @@
+                    "StrictHostKeyChecking=no",
+                    "-o",
+                    "UserKnownHostsFile /dev/null",
+                    "${config:torizon_run_as}@${config:torizon_ip}" // user@device
```
would be great to remove every comment from JSON files.

andreriesco replied:
Sounds good, I will remove it

microhobby commented on cppSlint/Dockerfile:
```diff
@@ -104,8 +104,7 @@ RUN apt-get -y update && apt-get install -y --no-install-recommends \
 	&& apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/*
 
-# Default to the Skia backend for best performance
-ENV SLINT_BACKEND=linuxkms-skia
```
Wait, why?

andreriesco replied:
Not needed anymore, comes from [here](https://github.com/torizon/vscode-torizon-templates/pull/268#discussion_r1809424607)

microhobby commented on DEPRECATED.md:
```diff
@@ -1,5 +1,20 @@
 # Deprecated templates
 
+## Deprecated on Torizon OS 7
```
I think that is important to highlight, or make a division between Community and Toradex supported ones. Because this could send a message that Toradex is deprecating, but this was a decision from community maintainer.

andreriesco replied:
Hmm ok, I will explain it better

andreriesco replied:
Detailed it a bit better now [here](https://github.com/torizon/vscode-torizon-templates/pull/268/commits/ca0b3705999bc9449aa9392e6aa8904aee666b04#diff-a82dba25f56e719cef404e54d5447f38b3c5571749ef3e0d0898da452886c83fR14)

microhobby replied:
@andreriesco the link https://github.com/torizon/vscode-torizon-templates/pull/268/commits/ca0b3705999bc9449aa9392e6aa8904aee666b04#diff-a82dba25f56e719cef404e54d5447f38b3c5571749ef3e0d0898da452886c83fR14 appears to be broken

microhobby replied:
oh, weird if I click on your [here](https://github.com/torizon/vscode-torizon-templates/pull/268/commits/ca0b3705999bc9449aa9392e6aa8904aee666b04#diff-a82dba25f56e719cef404e54d5447f38b3c5571749ef3e0d0898da452886c83fR14) it does not work, but If I copy the markdown link and only paste it here https://github.com/torizon/vscode-torizon-templates/pull/268/commits/ca0b3705999bc9449aa9392e6aa8904aee666b04#diff-a82dba25f56e719cef404e54d5447f38b3c5571749ef3e0d0898da452886c83fR14 it works.

microhobby commented on cspell.json:
```diff
@@ -14,6 +14,7 @@
         "Fleetid",
         "listenport",
         "netsh",
+        "notok",
```
please drop this commit 929506458af2b280a28e16131cddc872c905a427
As we discussed internally this change will be, maybe, added in a patch version before TOS 7

andreriesco replied:
Ok, once the patch get's merged I will rebase and then drop this one

microhobby commented on scripts/createFromTemplate.ps1:
```diff
@@ -216,8 +216,6 @@ $_metadataJson = New-Object PSObject
 $_metadataJson | Add-Member -MemberType NoteProperty -Name "templateName" -Value $template
 $_metadataJson | Add-Member -MemberType NoteProperty -Name "containerName" -Value $containerName
-
-$_templatesJson = Get-Content "$templateFolder/templates.json" | ConvertFrom-Json
```
🤔

andreriesco replied:
This is to get the TorizonOSMajor property

microhobby replied:
yah, but this was removed? My point is, this was introduced where to be already removed?

andreriesco replied:
It [wasn't](https://github.com/torizon/vscode-torizon-templates/pull/268/files#diff-7962b06fd40f0fb3e868d57cdcf3fab3cff59d583a6251b888dc102112447c2bR137), I think it's just the git presentation

microhobby replied:
this does not make sense. Because this was added here: 8bfa233
So, this is a fix for the mistake added on 8bfa233 ?

andreriesco replied:
It wasn't removed, it was just moved up on the script in this commit: [https://github.com/torizon/vscode-torizon-templates/pull/268/commits/7ad433868f32393952b00534bddc321eb0a11f5d#diff-e99af3c3a7d203492cf4f3596331f89f77804cb7efe89d84bdab0ace71278851R81](https://github.com/torizon/vscode-torizon-templates/pull/268/commits/7ad433868f32393952b00534bddc321eb0a11f5d#diff-e99af3c3a7d203492cf4f3596331f89f77804cb7efe89d84bdab0ace71278851R81)

Commit:
zigConsole: Fix Dockerfile

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco commented on zigConsole/Dockerfile:
```diff
@@ -17,45 +17,21 @@ ARG APP_ROOT=
 
 # BUILD ------------------------------------------------------------------------
-FROM torizon/cross-toolchain-${IMAGE_ARCH}:${SDK_BASE_VERSION} AS build
+FROM kassany/bookworm-ziglang:${SDK_BASE_VERSION} AS build
```
Hello @kassane,
When running the `Dockerfile` in the way it was, I was running into the following issue during build:
```
...
 => [zigTesr build 7/9] RUN rm -rf /home/torizon/app/target                                                                                                                                                  0.1s 
 => ERROR [zigtest build 8/9] RUN if [ "arm64" = "arm64" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=aarch64-linux-gnu --summary all -freference-trace ;     elif [ "arm64" = "armhf" ] ;   49.9s 
------                                                                                                                                                                                                           
 > [zigtest build 8/9] RUN if [ "arm64" = "arm64" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=aarch64-linux-gnu --summary all -freference-trace ;     elif [ "arm64" = "armhf" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=arm-linux-gnu --summary all -freference-trace ;     elif [ "arm64" = "amd64" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=x86_64-linux-gnu --summary all -freference-trace ;     fi:                                                                                                                                                                                      
49.71 Illegal instruction at address 0x1156abc                                                                                                                                                                   
49.71 Panicked during a panic. Aborting.                                                                                                                                                                         
49.72 qemu: uncaught target signal 6 (Aborted) - core dumped
49.82 error: the following build command crashed:
49.82 /home/torizon/app/.zig-cache/o/94cec820d9640d024730db2cf8a14d9f/build /zig/0.13.0/files/zig /home/torizon/app /home/torizon/app/.zig-cache /root/.cache/zig --seed 0xc9a9e621 -Z3a3bc4625ad946be -Doptimize[+] Building 0/1target=aarch64-linux-gnu --summary all -freference-trace
 ⠙ Service zigtest  Building                                                                                                                                                                                52.1s 
failed to solve: process "/bin/sh -c if [ \"$IMAGE_ARCH\" = \"arm64\" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=aarch64-linux-gnu --summary all -freference-trace ;     elif [ \"$IMAGE_ARCH\" = \"armhf\" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=arm-linux-gnu --summary all -freference-trace ;     elif [ \"$IMAGE_ARCH\" = \"amd64\" ] ; then         zig build -Doptimize=ReleaseSmall -Dtarget=x86_64-linux-gnu --summary all -freference-trace ;     fi" did not complete successfully: exit code: 1

 *  The terminal process "/usr/bin/bash '-c', 'DOCKER_HOST= docker compose build --pull --build-arg SSHUSERNAME=torizon --build-arg APP_ROOT=/home/torizon/app --build-arg IMAGE_ARCH=arm64 --build-arg GPU=-imx8 zigtest'" terminated with exit code: 17. 
 ```
 So I updated the base container of the build step to use [the same one as in the `Dockerfile.sdk`](https://github.com/torizon/vscode-torizon-templates/pull/268/commits/fa3798c929150eb62dda8e2196310f3bebca1a61#diff-5cc2e94a1d4553f577a2f04274575996ab4facdb181c243a5c05680d61648cacR18) and it worked

Commit:
common.json: Fix arm 32 SDK tasks for Torizon OS 7

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cppdbg: Remove engineLogging comment on launch.json files

This is to keep JSON files without comments

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cppConsole: Add build-container-image-sdk- as dependsOn of clean tasks

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
base containers: Use just the major 4 on the base containers

As agreed with the containers team, on Torizon 7 the containers will
need to pass also on the Torizon IDE Extension tests before any release,
ensuring their stability on the IDE templates. And, by not using a
<Major>.<Minor>.<Patch> and just <Major>, the user can get the newest
stable version of the base container on his template.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>
